src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -876,14 +882,18 @@ pub struct QemuCmdLine<'a> {
     smp: Smp,
     machine: Machine,
     cpu: Cpu,
-
+    bus_type: String,
```

Comment:
If this needs to be stored then it should live in Knobs. Alternatively, it could even be just a helper function in QemuCmdLine, something like

```
fn bus_type<'a>(&self) -> &'a str {
    return if self.config.machine_info.machine_type == "s390-ccw-virtio" {
          "ccw"
    } else {
          "pci"
    }
}
```
(NB pseudocode, didn't try to compile) which would keep our data structures simple.

Reply:
IMO, The type `bus_type:String` is not restrictive. I prefer to use an enum type to express `bus_type`.

Reply:
I followed Pavel's suggestion using `Enum`. Thanks!